pkg/kubelet/status/state/checkpoint.go

```diff
@@ -46,6 +46,14 @@ func NewPodResourceAllocationCheckpoint() *PodResourceAllocationCheckpoint {
 func (prc *PodResourceAllocationCheckpoint) MarshalCheckpoint() ([]byte, error) {
 	// make sure checksum wasn't set before so it doesn't affect output checksum
 	prc.Checksum = 0
+	data, err := json.Marshal(*prc)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(data, prc)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This looks like a workaround to me. If you think `Quantity.Unmarshal ` doesn't un-marshal quantity correctly, you should fix `Quantity.Unmarshal`.

Reply:
Precisely speaking, the un-marshal function isn't incorrect, but the behaviors of function marshal and un-marshal are different: the marshal function will convert quantity to `Canonical Bytes`, but un-marshal function won't. Honestly speaking, I'm not sure whether we should fix `Quantity.Unmarshal`.

Reply:
should we marshal -> unmarshal before calculating the checksum on write?

Reply:
@SergeyKanzhelev it works, but it more likes a workaround.

Reply:
@SergeyKanzhelev I'm not sure about it. Both solutions can fix the issue